test/integration/testManualConnection.js

```diff
@@ -259,4 +260,105 @@ if (process.env.RUN_MANUAL_TESTS_ONLY == "true") {
       });
     });
   });
+
+  describe.only('keepAlive test', function () {
+    this.timeout(180_000);
+    let connection;
+    const loopCount = 100;
+    const rowCount = 10;
+    const tableName = 'test_keepalive000';
+
+    const createTableWithRandomStrings = `CREATE OR REPLACE TABLE ${tableName} (value string)
+    AS select randstr(200, random()) from table (generator(rowcount =>${rowCount}))`;
+
+    before(async () => {
+      connection = snowflake.createConnection(connOption.valid);
+      await testUtil.connectAsync(connection);
+      await testUtil.executeCmdAsync(connection, createTableWithRandomStrings);
+    });
+    after(async () => {
+      await testUtil.dropTablesIgnoringErrorsAsync(connection, [tableName]);
+      await testUtil.destroyConnectionAsync(connection);
+    });
+
+    it('Verify that requests working faster with keep alive', function (done) {
+      let sumWithKeepAlive = 0;
+      let sumWithoutKeepAlive = 0;
+      async.series(
+        [
+          // Run the query loopCount times
+          async function () {
+            for (let count = 1; count <= loopCount; count++) {
+              await new Promise((resolve, reject) => {
+                const start = performance.now();
+                connection.execute({
+                  sqlText: `SELECT VALUE
+                            from ${tableName} limit ${rowCount};`,
+                  streamResult: true,
+                  complete: function (err, stmt) {
+                    if (err) {
+                      done(err);
+                    } else {
+                      stmt.streamRows()
+                        .on('error', function (err) {
+                          done(err);
+                        })
+                        .on('data', function (row) {
+                          return;
+                        })
+                        .on('end', function (row) {
+                          const end = performance.now();
+                          const time = end - start;
+                          sumWithKeepAlive += time;
+                          resolve();
+                        });
+                    }
+                  }
+                });
+              });
+            }
+          },
+          async function () {
+            snowflake.configure({keepAlive: false});
+            connection = snowflake.createConnection(connOption.valid);
+            await testUtil.connectAsync(connection);
```

Comment:
both test cases have the same test body - can we create single method to extract common things?